docs_src/security/Ch-SecretStore.md

```diff
@@ -17,1115 +29,259 @@ that you want to tightly control access to, such as API keys, passwords,
 
-Vault server configuration is essentially concentrated in one JSON file
-named `local.json`. This file was prepared during the Vault Docker image
-build process. In the eventuality of a change, the Vault server
-container should be accessed to then modify the JSON file. The absolute
-path being `/vault/config/local.json`. To reload the new configuration
-simply send Vault PID a HUP signal to trigger a configuration reload.
+## Using the Secret Store
 
-Sample Vault server configuration file:
+### Preferred Approach
 
-``` conf
-listener "tcp" {
-address = "edgex-vault:8200"
-tls_disable = "0"
-cluster_address = "edgex-vault:8201"
-tls_min_version = "tls12"
-tls_client_ca_file ="/vault/config/pki/EdgeXFoundryCA/EdgeXFoundryCA.pem"
-tls_cert_file ="/vault/config/pki/EdgeXFoundryCA/edgex-vault.pem"
-tls_key_file = "/vault/config/pki/EdgeXFoundryCA/edgex-vault.priv.key"
-}
+The preferred approach for interacting with the EdgeX secret store is to use the
+`SecretClient` interface in [go-mod-secrets](https://github.com/edgexfoundry/go-mod-secrets/blob/master/secrets/interfaces.go).
 
-backend "consul" {
-path = "vault/"
-address = "edgex-core-consul:8500"
-scheme = "http"
-redirect_addr = "https://edgex-vault:8200"
-cluster_addr = "https://edgex-vault:8201"
-}
+Each EdgeX microservice has access to a `StoreSecrets()` method that allows
+setting of per-microservice secrets, and a `GetSecrets()` method to read them back.
 
-default_lease_ttl = "168h"
-max_lease_ttl = "720h"
-```
+If manual "super-user" to the EdgeX secret store is required,
+it is necesary to obtain a privileged access token, called the Vault root token.
 
-The `listener` clause refers to Vault server process (port, TLS and
-server name), the `backend` clause refers to the storage backend (i.e.
-Consul).
+### Obtaining the Vault Root Token
 
-To modify this configuration file, execute a shell session in the
-running Vault container:
+For security reasons (the 
+[Vault production hardening guide](https://learn.hashicorp.com/tutorials/vault/production-hardening)
+recommends revokation of the root token), the Vault root token is revoked by default.
+EdgeX automatically manages the secrets required by the framework,
+and provides a programmatic interface for individual microservices
+to interact with their partition of the secret store.
 
-``` bash
-sh> docker exec -it -e PS1='\u@\h:\w \$ ' -e VAULT_CAPATH='/vault/config/pki/EdgeXFoundryCA/EdgeXFoundryCA.pem' edgex-vault sh
+If global access to the secret store is required,
+it is necessary to obtain a copy of the Vault root token
+using the below recommended procedure.
+Note that following this procedure directly contradicts the
+[Vault production hardening guide](https://learn.hashicorp.com/tutorials/vault/production-hardening).
+Since the root token cannot be un-revoked the framework
+must be started for the first time with root token revokation disabled.
```

Comment:
```suggestion
Since the root token cannot be un-revoked by the framework, which
must be started for the first time with root token revokation disabled.
```

Reply:
I meant  "Since the root token cannot be un-revoked, the framework
must be started for the first time with root token revokation disabled."